src/pubsub.js

```diff
@@ -1,162 +1,119 @@
 'use strict'
 
-const { multicodec: floodsubMulticodec } = require('libp2p-floodsub')
-const Pubsub = require('libp2p-pubsub')
-const pull = require('pull-stream')
-const lp = require('pull-length-prefixed')
-const nextTick = require('async/nextTick')
-const { constants: multistreamConstants } = require('multistream-select')
-const { utils } = require('libp2p-pubsub')
-const asyncMap = require('async/map')
+const assert = require('assert')
+const debug = require('debug')
+const debugName = 'libp2p:gossipsub'
+const log = debug(debugName)
+log.error = debug(`${debugName}:error`)
```

Comment:
Why aren't we using `this.log` here? Maybe indirectly causing one of the tests to fail.

Reply:
So, the issue with using `this.log` is that when we are in the scope of a `for await ...` loop, we can't use `this.x` in that way. I am still looking for a better way of doing this, let's see if I can get a better way

Reply:
I'm not sure I understand. I don't see any logs inside `for await` loops.
My impression is that `this` stays the same even inside that scope. Is there some sort of tricky case where thats not the case?

Reply:
I could get this working by using bing